infra/token-transfer-client/src/components/LockupCard.js

```diff
@@ -43,7 +43,11 @@ const LockupCard = ({ lockup }) => {
           ></div>{' '}
           Bonus tokens
           <br />
-          <strong>{Number(lockup.amount).toLocaleString()}</strong>{' '}
+          <strong>
+            {Math.floor(
+              Number((lockup.amount * lockup.bonusRate) / 100)
```

Comment:
1. Should this use BigNumber for this math. I don't trust JS floating point arithmetic...
2. Why do we floor? Can't the bonus token amount be decimal? For example if the user locks 123 tokens and the bonus rate is 10%, should the user get 12.3 bonus tokens or 12?

Reply:
1. Will change it to BN 👍
2. Because we made the decision to use only integers for token handling everywhere else in T3. For example withdrawals don’t allow decimals, and we round vesting amounts to integers too. We could allow decimals here, probably not too many negatives that I can think of (except not being able to withdraw decimal amounts). What do you think?

Reply:
I'd vote for keeping it simple and rounding to integer.
Though imho we should round up rather than round down-let's not be cheap, it's only a few cents we are talking about...